src/components/calcite-date-picker/calcite-date-picker.tsx

```diff
@@ -0,0 +1,495 @@
+  /** Selected date */
+  @Prop() value?: string;
+
+  /** Selected date as full date object*/
+  @Prop({ mutable: true }) valueAsDate?: Date;
+
+  /** Selected start date as full date object*/
+  @Prop({ mutable: true }) startAsDate?: Date;
+
+  /** Selected end date as full date object*/
+  @Prop({ mutable: true }) endAsDate?: Date;
+
+  /** Earliest allowed date ("yyyy-mm-dd") */
+  @Prop() min?: string;
+
+  /** Latest allowed date ("yyyy-mm-dd") */
+  @Prop() max?: string;
+
+  /** Localized string for "previous month" (used for aria label) */
+  @Prop() intlPrevMonth?: string = TEXT.prevMonth;
+
+  /** Localized string for "next month" (used for aria label) */
+  @Prop() intlNextMonth?: string = TEXT.nextMonth;
+
+  /** BCP 47 language tag for desired language and country format */
+  @Prop() locale?: string = document.documentElement.lang || "en-US";
+
+  /** specify the scale of the date picker */
+  @Prop({ reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** Range mode activation */
+  @Prop({ reflect: true }) range?: boolean = false;
+
+  /** Selected start date */
+  @Prop() start?: string;
+
+  /** Selected end date */
+  @Prop() end?: string;
+
+  @Prop() proximitySelection?: boolean = true;
+
+  /** Layout */
+  @Prop({ reflect: true }) layout: "horizontal" | "vertical" = "horizontal";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+  @Listen("blur")
+  focusOutHandler(): void {
+    this.reset();
+  }
+
+  /**
+   * Blur doesn't fire properly when there is no shadow dom (ege/IE11)
+   * Check if the focused element is inside the date picker, if not close
+   */
+  @Listen("focusin", { target: "window" })
+  focusInHandler(e: FocusEvent): void {
+    if (!this.hasShadow && !this.el.contains(e.srcElement as HTMLElement)) {
+      this.reset();
+    }
+  }
+
+  @Listen("keyup")
+  keyDownHandler(e: KeyboardEvent): void {
+    if (getKey(e.key) === "Escape") {
+      this.reset();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+  /**
+   * Trigger calcite date change when a user changes the date.
+   */
+  @Event() calciteDatePickerChange: EventEmitter<Date>;
+
+  /**
+   * Trigger calcite date change when a user changes the date range.
+   */
+  @Event() calciteDatePickerRangeChange: EventEmitter<DateRangeChange>;
+
+  /**
+   * Active date.
+   */
+  @State() activeDate: Date;
+
+  /**
+   * Active start date.
+   */
+  @State() activeStartDate: Date;
+
+  /**
+   * Active end date.
+   */
+  @State() activeEndDate: Date;
+
+  // todo: figure out a better API prop for this?
+  /**
+   * In range mode, indicates which input was is focused on
+   */
+  @State() focusedInput: "start" | "end" = "start";
```

Comment:
I think this needs to be changed to not have anything to do with "input" but more of a "state". Either entering a start or end date.

Reply:
Like a picking state.